src/VisualStudio/Core/Def/Implementation/FindReferences/StreamingFindUsagesPresenter.cs

```diff
@@ -176,6 +176,10 @@ public FindUsagesContext StartSearch(string title, bool supportsReferences, Canc
             // Get the appropriate window for FAR results to go into.
             var window = vsFindAllReferencesService.StartSearch(title);
 
+            // Wrap the passed in CT with our own CTS that we can control cancellation over.  This way either our
+            // caller can cancel our work or we can cancel the work.
+            var cancellationTokenSource = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
```

Comment:
❓ Where does this get disposed (cancellation token sources created by `CreateLinkedTokenSource` must be disposed or they will leak memory)

Reply:
ah.  the presenter is disposable.  i will add.  thanks!

Reply:
so we do properly already dispose thsi.  i will add comment here about the ownership going on here.